SCClassLibrary/Common/Audio/BasicOpsUGen.sc

```diff
@@ -171,8 +171,9 @@ BinaryOpUGen : BasicOpUGen {
 
 		if (b.isKindOf(UnaryOpUGen) and: { b.operator == 'neg' and: { b.descendants.size == 1 } }) {
 			// a + b.neg -> a - b
+			if(b === a) { ^nil };  // non optimizable edge case.
 			buildSynthDef.removeUGen(b);
-			replacement = a - b.inputs[0];
+			replacement = BinaryOpUGen('-', a, b.inputs[0]);
```

Comment:
While this change isn't harmful, it doesn't really improve anything either. I would leave this, and the next occurrence, alone.

Reply:
I thought it as proposal for review. It makes explicit use of the replacement object as done below by other method which also makes apparently dissimilar code (and the op gets hide to the view too). Can I keep it?